### PR TITLE
Simplify tests

### PR DESCRIPTION
**Pull request by mvsusp**
- not reloading module anymore to avoid inheritance errors
- remove dependency from keras in the tests
- simplified tests

It is the same PR that @winstonaws and @iquintero approved before, plus the keras removal and fix for the loading modules issue.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker_containers/environment.py [hunk 1]
```diff
@@ -37,6 +40,7 @@
 BASE_PATH_ENV = 'BASE_PATH'  # type: str
 CURRENT_HOST_ENV = 'CURRENT_HOST'  # type: str
 JOB_NAME_ENV = 'JOB_NAME'  # type: str
+USE_NGINX_ENV = 'SAGEMAKER_USE_NGINX'  # type: str
```
Comment (iquintero) [inline src/sagemaker_containers/environment.py]: ? this is already defined below. Seems like the merge was not clean.

File: src/sagemaker_containers/environment.py [hunk 2]
```diff
@@ -51,6 +55,11 @@
 RESOURCE_CONFIG_FILE = 'resourceconfig.json'  # type: str
 INPUT_DATA_CONFIG_FILE = 'inputdataconfig.json'  # type: str
 
+HYPERPARAMETERS_PATH = os.path.join(INPUT_CONFIG_PATH, HYPERPARAMETERS_FILE)  # type: str
+INPUT_DATA_CONFIG_FILE_PATH = os.path.join(INPUT_CONFIG_PATH, INPUT_DATA_CONFIG_FILE)  # type: str
+RESOURCE_CONFIG_PATH = os.path.join(INPUT_CONFIG_PATH, RESOURCE_CONFIG_FILE)  # type: str
+
+PROGRAM_PARAM = 'sagemaker_program'  # type: str
```
Comment (iquintero) [inline src/sagemaker_containers/environment.py]: same as above, USER_PROGRAM_PARAM is defined just below this line.

File: src/sagemaker_containers/environment.py [hunk 3]
```diff
@@ -679,3 +689,24 @@ def model_server_workers(self):  # type: () -> int
         """Returns:
             (int): Number of worker processes the model server will use"""
         return self._model_server_workers
+
+
+@contextlib.contextmanager
+def temporary_directory(suffix='', prefix='tmp', dir=None):  # type: (str, str, str) -> None
```
Comment (iquintero) [inline src/sagemaker_containers/environment.py]: should this temporary_directory live in environment? I dont think so, this seems to be a generic context manager for temp dirs and I can see that its used at least in modules.py as well. Maybe functions.py?

File: test/miniml.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (iquintero) [inline test/miniml.py]: minimal.py? or is miniml.py intended?

File: test/unit/test_environment.py [hunk 5]
```diff
@@ -20,7 +20,7 @@
 import six
 
 import sagemaker_containers as smc
-from test.conftest import json_dump
+import test
```
Comment (iquintero) [inline test/unit/test_environment.py]: Can we not use test?

> 
> The test package is meant for internal use by Python only. It is documented for the benefit of the core developers of Python. Any use of this package outside of Python’s standard library is discouraged as code mentioned here can change or be removed without notice between releases of Python.
> https://docs.python.org/3.5/library/test.html

File: src/sagemaker_containers/environment.py (hunk 1 above)
Comment (mvsusp) [reply src/sagemaker_containers/environment.py]: I will fix it.

File: src/sagemaker_containers/environment.py (hunk 2 above)
Comment (mvsusp) [reply src/sagemaker_containers/environment.py]: I will fix it.

File: src/sagemaker_containers/environment.py (hunk 3 above)
Comment (mvsusp) [reply src/sagemaker_containers/environment.py]: Yeah, I thought about it. My thought process was that temporary files are part of the environment. Functions namespace only include functions that deal with functions. Do you have another suggestion?

Comment (laurenyu) [inline src/sagemaker_containers/environment.py]: maybe `tmp_dir` or `tmpdir` for the method name instead? or `create_tmpdir` if you're worried about overusing 'tmpdir'

File: test/miniml.py (hunk 4 above)
Comment (mvsusp) [reply test/miniml.py]: It is a mock name for a framework. Do you prefer mini_ml ? Or mocked_framework ?

Comment (iquintero) [reply test/miniml.py]: ah, mini ml lol that works

File: src/sagemaker_containers/environment.py (hunk 2 above)
Comment (iquintero) [reply src/sagemaker_containers/environment.py]: You missed this one.

Comment (mvsusp) [reply src/sagemaker_containers/environment.py]: I missed because they don't have the same name. Sorry about that.

**Pull request merged**